iothub/service/tests/QueryTests.cs

```diff
@@ -292,5 +293,58 @@ public void QueryResultUserSuppliedContinuationTest()
             Assert.AreEqual("test", r.ElementAt(0).DeviceId);
             Assert.IsTrue(q.HasMoreResults);
         }
+
+        [TestMethod]
+        public void QueryResult_OverrideDefaultJsonSerializer_Ok()
+        {
+            // simulate json serialize/deserialize
+            var serverQueryResult = new QueryResult
+            {
+                Type = QueryResultType.Twin,
+                Items = new List<Twin>
+                {
+                    new Twin
+                    {
+                        DeviceId = "test",
+                    }
+                },
+                ContinuationToken = "GYUVJDBJFKJ"
+            };
+
+            var settings = new JsonSerializerSettings { MaxDepth = 3 };
+            // serialize
+            var jsonQueryResult = JsonConvert.SerializeObject(serverQueryResult, settings);
+            jsonQueryResult.Should().Be("{\"type\":\"twin\",\"items\":[{\"deviceId\":\"test\",\"etag\":null,\"version\":null,\"properties\":{\"desired\":{},\"reported\":{}}}],\"continuationToken\":\"GYUVJDBJFKJ\"}");
+            // deserialize
+            var clientQueryResult = JsonConvert.DeserializeObject<QueryResult>(jsonQueryResult, settings);
+        }
+
+        [TestMethod]
+        public void QueryResult_OverrideDefaultJsonSerializer_ExceedMaxDepthThrows()
+        {
+            // simulate json serialize/deserialize
+            var serverQueryResult = new QueryResult
+            {
+                Type = QueryResultType.Twin,
+                Items = new List<Twin>
+                {
+                    new Twin
+                    {
+                        DeviceId = "test",
+                    }
+                },
+                ContinuationToken = "GYUVJDBJFKJ"
+            };
+
+            var settings = new JsonSerializerSettings { MaxDepth = 2 };
+            // serialize
+            var jsonQueryResult = JsonConvert.SerializeObject(serverQueryResult, settings);
```

Comment:
Isn't the serialize call expected to throw an exception as well?

Reply:
Looks like the max depth limit is only enforced when reading JSON data, not when writing it.

Reply:
You're right about that: https://www.newtonsoft.com/json/help/html/P_Newtonsoft_Json_JsonSerializerSettings_MaxDepth.htm
"Gets or sets the maximum depth allowed when reading JSON. Reading past this depth will throw a [JsonReaderException](https://www.newtonsoft.com/json/help/html/T_Newtonsoft_Json_JsonReaderException.htm)."

You can remove the serialization bit from the test case.

Reply:
Sure!